### PR TITLE
fix(deps): update React types and fix csstype errors

### PR DESCRIPTION
**Pull request by denkristoffer**
# Purpose of PR

This PR ensures that the same React types are used in all components and fixes the `csstype` errors this lead to. It also adds `skipLibCheck: true` to the `tsconfig.json` in `forma-36-react-components` to avoid errors from `@types/react` itself.

Is this a `fix` or is it bigger than that? 🤔 

## PR Checklist

- [x] I have read the relevant `readme.md` file(s)
- [x] All commits follow our [Git commit message convention](https://github.com/contentful/forma-36/tree/master/packages/forma-36-react-components#commits)
- [x] Tests are added/updated/not required
- [x] Tests are passing
- [x] Storybook stories are added/updated/not required
- [x] Usage notes are added/updated/not required
- [x] Has been tested based on [Contentful's browser support](https://www.contentful.com/faq/about-contentful/#which-browsers-does-contentful-support)
- [x] Doesn't contain any sensitive information

### CODE REVIEW

**Review by gui-santos — COMMENTED** [review]
Uh I don't know
It seems ok to me, curious about what the others think 👀

**Review by mshaaban0 — COMMENTED** [review]
I think it's just the `css-type` syntax that is different, the type values remains the same.

Not so sure about `skipLibCheck` though, does that mean `@contentful/forma-36-tokens` and other packages are skipped if they contain type errors as well?

File: packages/components/forma-36-react-timepicker/package.json [hunk 1]
```diff
@@ -33,8 +33,8 @@
     "@types/is-hotkey": "^0.1.1",
     "@types/jest": "^24.0.22",
     "@types/lodash.orderby": "^4.6.6",
-    "@types/react": "^16.9.11",
-    "@types/react-dom": "^16.9.3",
+    "@types/react": "^16.9.56",
+    "@types/react-dom": "^16.9.9",
```
Comment (mshaaban0) [inline packages/components/forma-36-react-timepicker/package.json:37]: do you think we should add `csstype` as a dependency, so we control the version and avoid relying on it from react?

Comment (denkristoffer) [reply packages/components/forma-36-react-timepicker/package.json:37]: Yeah good point, we should when we're using it directly anyway.

**Comment by denkristoffer** [comment]
> Not so sure about skipLibCheck though, does that mean @contentful/forma-36-tokens and other packages are skipped if they contain type errors as well?

I think so, but we're building them separately and type checking them in that process, no?

**Review by gui-santos — APPROVED** [review]
LGTM

**Pull request merged**